### DISCUSSION TITLE
Github Actions: Branch filter not working

### DISCUSSION THREAD

**Discussion by phil9690** [community/community#25838] [Actions]
<p>I’m trying to trigger some actions only when a branch starts with a specific pattern, but it’s not working. When I remove the filter it works, but obviously will do it on all pull requests, which I don’t want. In my case here I’d expect any branch that starts with ‘branch_name_’ will trigger the actions.</p>
<pre><code>name: GAC UI Release
on:
  pull_request:
    types: [opened]
    branches:
    - 'branch_name_*'
</code></pre>
<p><a href="https://help.github.com/en/articles/workflow-syntax-for-github-actions#onpushpull_requestbranchestags" rel="nofollow noopener">https://help.github.com/en/articles/workflow-syntax-for-github-actions#onpushpull_requestbranchestags</a></p>
<p>The docs here suggest this should work, unless I’m missing something.</p>

**Comment by chrispat** [comment, marked as answer]
<p>We have shipped a fairly significant update to our filtering syntax for events <a href="https://help.github.com/en/articles/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet" rel="nofollow noopener">https://help.github.com/en/articles/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet</a>.  I hope this resolves your issue.  If not please let us know.</p>

**Comment by ArnaudRinquin** [comment]
<p>Works 100% better now, thanks for this update!</p>

**Comment by phil9690** [comment]
<p><a class="mention" href="/chrispat">@chrispat</a>  I’m still having no luck with this. I’ve tried these patterns:</p>
<pre><code>'branch_name_*'
'branch_name_**'
'branch_name*'
'branch_name**'
</code></pre>
<p>Here is an example of branch ref:  <strong>refs/heads/branch_name_v1.10_foo</strong></p>
<p>Since the branch starts with <strong>branch_name </strong> I would expect all the patterns I’ve mentioned above to work.</p>
<p>If I do a branch filter on only  <strong>‘*’</strong>  the actions will all run without any errors.</p>
<p>Any ideas on what could be going wrong here?</p>

**Comment by chrispat** [comment]
<p>I am not able to reporduce that issue.  Perhaps there is some confusion about how the branches filter works in pull requests.  In pull_requests you are specifying the name of the branch the pull request is targeting or the base branch.  The idea is that you want to have a workflow that validates all changes that someone is trying to merge into a branch you want to protect.</p>

**Comment by phil9690** [comment]
<p><a class="mention" href="/chrispat">@chrispat</a>   thanks, that makes sense. I assumed it was going work like <a href="https://github.com/actions/bin/tree/master/filter" rel="nofollow noopener">https://github.com/actions/bin/tree/master/filter</a>. I take it in my case I’d still need to use the branch filter action to trigger my actions if it passes, or is there another way to do this?</p>
<p>What I’m trying to do is check if a PR has been raised with a specific branch name targeting release, if it matches the name I’m looking for, then run my actions otherwise don’t do anything.</p>

**Comment by chrispat** [comment]
<p>At the moment there really is no good way to do that.  We have some additional features planned around filtering and conditionals for workflows that should enable you to do pretty much anything you want.</p>

**Comment by phil9690** [comment]
<p>Good to hear, thanks for the update.</p>

**Comment by phil9690** [comment]
<p>Hi <a class="mention" href="/chrispat">@chrispat</a>, it looks like <a href="https://github.com/actions/bin/tree/master/filter" rel="nofollow noopener">https://github.com/actions/bin/tree/master/filter</a> has been removed, so now I can’t run the branch filter I was previously using. Has this functionality now been incorporated into workflow filter patterns? I can’t see anything updated in the docs to say this filter can now work on the branch I’m raising the PR with and not the branch I’m targeting with my PR. Is there anything I can use for this now?</p>
